capstone/src/main/java/com/google/step/servlets/RestaurantServlet.java

```diff
@@ -32,43 +32,48 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet responsible for sending/getting restaurants to/from Datastore. */
+/** Servlet responsible for sending/getting a restaurant to/from Datastore. */
 @WebServlet("/restaurant")
 public class RestaurantServlet extends HttpServlet {
+  /**
+   * Returns a single Restaurant based on restaurantKey.
+   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("RestaurantInfo").addSort("score", SortDirection.ASCENDING);
-
+    Long restaurantKeyParam = Long.parseLong(request.getParameter("restaurantKey"));
```

Comment:
We should talk about auto-boxing and using boxed primitives. Boxed primitives are the "class" counterparts of primitives  (Long, Integer, etc.). In general, you should avoid using them unless absolutely necessary, especially when doing a computation. If you have access to Effective Java, it has an item about boxed primitives and auto-boxing.

That said, in THIS case, you're okay, because even though you're auto-boxing here (Long.parseLong returns a `long`, and you're autoboxing it into a `Long`), you're just passing that `Long` into a FilterPredicate, which seems to take an Object, so you're fine.

No change to make here, just wanted to call it out in general, in case it comes up later.

Reply:
Thanks for this information--I'll look into it in Effective Java as well, and be on the lookout for future code!